lib/electric_slide/call_queue.rb

```diff
@@ -89,6 +89,8 @@ def get_agent(id)
     # @param [Agent] agent The agent to be added to the queue
     def add_agent(agent)
       abort ArgumentError.new("#add_agent called with nil object") if agent.nil?
+      abort ArgumentError.new("Agent is already in the queue") if get_agent(agent.id)
```

Comment:
I'd prefer to raise an exception that we can filter on, like `ElectricSlide::CallQueue::DuplicateAgentError`.